sys/include/net/ng_pktbuf.h

```diff
@@ -57,33 +57,37 @@ extern "C" {
 #endif  /* NG_PKTBUF_SIZE */
 
 /**
+ * @brief   Initializes packet buffer module.
+ */
+void ng_pktbuf_init(void);
+
+/**
  * @brief   Adds a new ng_pktsnip_t and its packet to the packet buffer.
  *
- * @details This function is very powerful and reflects the unique characterics
- *          of ng_pktsnip_t of being reversed for either the sending or
- *          receiving context. Because of this the assumtion of the transmission
- *          direction, the state of the packet buffer and the values for the
- *          members of the resulting ng_pktsnip_t can be very different after
- *          execution of this function depending on what parameters you use:
- *
- * * The return value of this function is a @ref ng_pktsnip_t struct referred
- *   to as `result`
- * * for most cases the build-up of `result` will be pretty straight forward: Its
- *   members will be exactly as the given parameters (ng_pktsnip_t::next of
- *   result will be set to @p pkt). If @p pkt is not NULL it and in turn `result`
- *   are assumed to be in sending direction. For packet creation (@p pkt == NULL)
- *   no assumtions about direction of `result` will be made (since its
- *   ng_pktsnip::next will be set to NULL).
- * * if @p pkt != NULL, @p data = `pkt->data`, @p size < `pkt->size` receiving
- *   direction is assumed and the following values will be set:
- *   * ng_pktsnip_t::next of `result` = `pkt->next`
- *   * ng_pktsnip_t::data of `result` = @p data
- *   * ng_pktsnip_t::size of `result` = @p size
- *   * ng_pktsnip_t::next of @p pkt = `result`
- *   * ng_pktsnip_t::data of @p pkt = @p data + @p size
- *   * ng_pktsnip_t::size of @p pkt = old size value - @p size
- *
- *  graphically this can be represented as follows:
+ * @warning **Do not** change the ng_pktsnip_t::data and ng_pktsnip_t::size
+ *          of a ng_pktsnip_t created by this function externally, except if
+ *          they both are null or data is not from inside the packet buffer.
+ *          This will most likely create memory leaks.
```

Comment:
Can't we simplify this: '**Do not** change the fields of the ng_pktsnip_t created by this function externally. This will most likely create memory leaks or not allowed memory access.